src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    if (this.hidden) {
+      this.hidden = false;
+
+      return;
```

Comment:
don't see any sense for adding empty return here
```suggestion
```

Reply:
I've done it, because I should finish execution of method hide(). In other way, I will have this.hidden equals true again, after condition.
I could use "if.. else", but other mentors said don't do that in this way.

Reply:
method hide should only hide element, not toggle 
so, in this case you can write it like this:
```
hide() {
  if (!this.hidden) {
   this.hidden = true;
  }
}
```